### PR TITLE
Added averaging script for torch dist

### PR DESCRIPTION
**Pull request by jubick1337**
# What does this PR do ?

Adding script to run checkpoint averaging for torch distributed checkpoint format.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+def save_distributed_weights_dcp(ckpt_path, averaged_state_dict):
+    logging.info(f"Saving averaged checkpoint to {ckpt_path}")
+
+    os.makedirs(ckpt_path, exist_ok=True)
+    save(
+        sharded_state_dict=averaged_state_dict,
+        checkpoint_dir=ckpt_path,
+        sharded_strategy=TorchDistSaveShardedStrategy(backend="torch_dist", version=1),
+        validate_access_integrity=False,
```
Comment (mikolajblaz) [inline scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:123]: why is it False?

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+
+def save_distributed_weights_dcp(ckpt_path, averaged_state_dict):
+    logging.info(f"Saving averaged checkpoint to {ckpt_path}")
+
+    os.makedirs(ckpt_path, exist_ok=True)
+    save(
+        sharded_state_dict=averaged_state_dict,
+        checkpoint_dir=ckpt_path,
+        sharded_strategy=TorchDistSaveShardedStrategy(backend="torch_dist", version=1),
```
Comment (mikolajblaz) [inline scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:122]: You can just pass `sharded_strategy=torch_dist` and consequently make it configurable in the CLI (allowing e.g. 'zarr' if someone wants to have 'zarr' for some reason; it's deprecated but sometimes people fallback to it)

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        ckpt_name = os.path.dirname(ckpt_name)
+        shutil.copy(
+            os.path.join(args.untarred_nemo_folder, "model_config.yaml"),
+            os.path.join(ckpt_name, "model_config.yaml"),
+        )
+
+        files = glob.glob(f"{args.untarred_nemo_folder}/*.model") + glob.glob(
+            f"{args.untarred_nemo_folder}/*.vocab.json"
+        )
+        logging.info(f"Copying other files: {files}")
```
Comment (mikolajblaz) [inline scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:213]: Do we want to copy the original `common.pt` (which contains all the PTL training related data like params, loops state etc.)?

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py (hunk 2 above)
Comment (jubick1337) [reply scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:122]: There's a script for zarr format [here](https://github.com/NVIDIA/NeMo/blob/main/scripts/checkpoint_averaging/distributed_checkpoint_averaging.py).
I guess it can be merged into one, but so far, all of the formats had separate scripts

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py (hunk 1 above)
Comment (jubick1337) [reply scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:123]: I don't know, it worked perfectly so far.
Should I change it to `True`?

Comment (mikolajblaz) [reply scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:123]: yes please, I will help avoid potential issues in the future

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py (hunk 2 above)
Comment (mikolajblaz) [reply scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:122]: I think having one script instead of 2 will be better going forward, but I'm not a user of those scripts so maybe there are some details I'm missing.

We don't have to remove the other one, we could just add an option to this one for a smooth transition

File: scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py (hunk 3 above)
Comment (mikolajblaz) [reply scripts/checkpoint_averaging/torch_distributed_checkpoint_averaging.py:213]: If the averaged ckpts are used for inference/fine-tuning then probably not :+1:
